nomenclature/definition.py

```diff
@@ -136,22 +146,67 @@ def check_aggregate(self, df: IamDataFrame, **kwargs) -> None:
             error = pd.concat(lst)
             return error if not error.empty else None
 
-    def to_excel(
-        self, excel_writer, sheet_name=None, sort_by_code: bool = False, **kwargs
-    ):
-        """Write the *variable* codelist to an Excel sheet
+    def to_excel(self, excel_writer, sort_by_code: bool = False, **kwargs):
+        """Write the codelists to an xlsx spreadsheet
 
         Parameters
         ----------
         excel_writer : path-like, file-like, or ExcelWriter object
             File path as string or :class:`pathlib.Path`,
             or existing :class:`pandas.ExcelWriter`.
-        sheet_name : str, optional
-            Name of sheet that will have the codelist. If *None*, use the codelist name.
         sort_by_code : bool, optional
             Sort the codelist before exporting to file.
         **kwargs
             Passed to :class:`pandas.ExcelWriter` (if *excel_writer* is path-like).
         """
-        # TODO write all dimensions to the file
-        self.variable.to_excel(excel_writer, sheet_name, sort_by_code, **kwargs)
+        with pd.ExcelWriter(excel_writer, engine="xlsxwriter", **kwargs) as writer:
+
+            # create dataframe with attributes of the DataStructureDefinition
+            arg_dict = {
+                "project": self.working_dir.absolute().parts[-1],
+                "file created": time_format(datetime.now()),
+                "": "",
+            }
+            if self.repo is not None:
+                arg_dict.update(git_attributes("", self.repo))
+
+            ret = make_dataframe(arg_dict)
+
+            for key, value in self.config.repositories.items():
+                ret = pd.concat(
+                    [
+                        ret,
+                        make_dataframe(git_attributes(key, git.Repo(value.local_path))),
+                    ]
+                )
+
+            write_sheet(writer, "project", ret)
+
+            # write codelist for each dimensions to own sheet
+            for dim in self.dimensions:
+                getattr(self, dim).to_excel(writer, dim, sort_by_code)
+
+
+def time_format(x):
+    return x.strftime("%Y-%m-%d %H:%M:%S")
+
+
+def git_attributes(name, repo):
+    return {
+        "repository": name,
+        "url": repo.remote().url,
+        "commit": repo.commit(),
+        "timestamp": time_format(repo.commit().committed_datetime),

```

Comment:
Maybe change it to "last commit at" to make it more obvious what it means:
```suggestion
        "last commit at": time_format(repo.commit().committed_datetime),
```

Reply:
This is basically superseded by your later suggestion, I guess...

Reply:
True, good point.